tests/storage/mpathconf_test.py

```diff
@@ -102,3 +117,33 @@ def test_read_bad_blacklist(fake_conf):
     fake_conf.write(BAD_CONF)
     wwids = mpathconf.read_blacklist()
     assert wwids == {"wwid1", "wwid2", "wwid5"}
+
+
+@pytest.fixture
+def fake_mpath_conf(tmpdir, monkeypatch):
+    fake_conf = tmpdir.join("multipath.conf")
+    monkeypatch.setattr(mpathconf, "CONF_FILE", str(fake_conf))
+
+    def _write_data(data):
+        fake_conf.write(data)
+        return fake_conf
+
+    return _write_data
```

Comment:
Returning a function works but not consistent with other fixtures.

When we have a fixture that needs configuration, the best way to  implement it is
returning an object supporting attributes and methods that helps to create the fixture
content. This is the way pytest fixture are implemented and we follow this design.

The fixture for multiapth conf need to do 2 things:
- monkeypatch the path to the module will find the fake file
- write content - this must be done in the test context, since each test want to use different content.

The simplest way to implement this is to do:

    @pytest.fixture
    def fake_conf(tmpdir, monkeypatch):
        path = tmpdir.join("multipath.conf")
        monkeypatch.setattr(mpathconf, "CONF_FILE", str(path))
        return path

Now the test can write the needed content:

    def test_mpath_conf(fake_conf):
        fake_conf.write("test data")
        test it...